effectzgpt/backend/app/engine/generate.py

```diff
@@ -31,14 +32,20 @@ def get_doc_store():
 
 
 def run_pipeline(docstore, vector_store, documents):
-    pipeline = IngestionPipeline(
-        transformations=[
-            SentenceSplitter(
+    transformations = []
+    transformations.append(SentenceSplitter(
                 chunk_size=Settings.chunk_size,
                 chunk_overlap=Settings.chunk_overlap,
-            ),
-            Settings.embed_model,
-        ],
+            ))
+    transformations.append(Settings.embed_model)
+
+    if os.getenv("USE_SENTENCE_WINDOW_RETRIEVAL", "True").lower() == "true":
+        transformations.append(get_sentence_window_node_parser())
+    
+    transformations.append(get_sentence_window_node_parser())
```

Comment:
The same transformation is applied within the IF statement and outside the IF statement.

Reply:
Thank you for noticing. Will remove the second statement.

Reply:
Requested change have been implemented @iujinasena pls review.